### ISSUE TITLE
equihash unable to benchmark and run on GTX660m

### ISSUE THREAD

**Issue by cjlcjlcjl43** [nicehash/NiceHashMiner-Archived#322]
Using 1.7.3.3 and 375.70 driver,tried all the 3 types of benchmark. quick, standard and precise. 
Speed showing none.Can you help?

**Comment by cjlcjlcjl43** [comment]
[2016-11-04 22:43:58] [INFO] [ComputeDeviceQueryManager] QueryCPUs START
[2016-11-04 22:43:58] [INFO] [ComputeDeviceQueryManager] QueryCPUs END
[2016-11-04 22:43:58] [INFO] [ComputeDeviceQueryManager] QueryCudaDevices START
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] 
CudaDevicesDetection:
    ADDED device:
        ID: 0
        NAME: V_ID_5464 GeForce GTX 660M
        VENDOR: V_ID_5464
        UUID: GPU-8078e1f9-62fe-bee0-ac55-eb3be0e643bb
        SM: 3.0
        MEMORY: 2147483648
        ETHEREUM: YES

[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] QueryCudaDevices END
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] QueryOpenCLDevices START
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] 
AMDOpenCLDeviceDetection found devices success:
    Found devices for platform: Intel(R) OpenCL
        Device:
            Device ID 0
            Device NAME Intel(R) HD Graphics 4000
            Device TYPE GPU
    Found devices for platform: NVIDIA CUDA
        Device:
            Device ID 0
            Device NAME GeForce GTX 660M
            Device TYPE GPU

[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] QueryOpenCLDevices END
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] QueryAMD START
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] Checking AMD device (driver): Intel(R) HD Graphics 4000 (10.18.10.4358)
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] Checking AMD device (driver): NVIDIA GeForce GTX 660M (21.21.13.7570)
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] QueryAMD END
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] Cuda NVIDIA/CUDA device count GOOD
[2016-11-04 22:44:01] [INFO] [ComputeDeviceQueryManager] AMD GPU device count GOOD
[2016-11-04 22:44:01] [INFO] [NICEHASH] Version get
[2016-11-04 22:44:01] [INFO] [MinersManager] Creating MinerPathChecker miners
[2016-11-04 22:44:01] [INFO] [AMD_OpenCL-MINER_ID(0)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:01] [INFO] [NVIDIA2.1-MINER_ID(1)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:01] [INFO] [NVIDIA3.x-MINER_ID(2)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:01] [INFO] [NVIDIA5.x-MINER_ID(3)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:01] [INFO] [NVIDIA6.x-MINER_ID(4)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:02] [INFO] [NICEHASH] SMA get
[2016-11-04 22:44:02] [INFO] [COINBASE] Bitcoin rate get
[2016-11-04 22:44:02] [INFO] [COINBASE] Current Bitcoin rate: 699.60
[2016-11-04 22:44:02] [INFO] [NICEHASH] Balance get
[2016-11-04 22:44:02] [INFO] [NICEHASH] Setting environment variables
[2016-11-04 22:44:02] [INFO] [NICEHASH] Trying to enable/disable Windows error reporting
[2016-11-04 22:44:02] [INFO] [NICEHASH] Current DontShowUI value: 1
[2016-11-04 22:44:07] [INFO] [nheqminer-MINER_ID(5)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:07] [INFO] [ExtraLaunchParametersParser] ExtraLaunch params "" for device UUID GPU-8078e1f9-62fe-bee0-ac55-eb3be0e643bb
[2016-11-04 22:44:07] [INFO] [ExtraLaunchParametersParser] Final extra launch params parse ""
[2016-11-04 22:44:07] [INFO] [BENCHMARK] Benchmark starts
[2016-11-04 22:44:07] [INFO] [nheqminer-MINER_ID(5)-DEVICE_IDs(0)] Starting benchmark: -b   -t 0  -cd 0  -cv 1 
[2016-11-04 22:44:07] [INFO] [nheqminer-MINER_ID(5)-DEVICE_IDs(0)] Using miner: bin\nheqminer_v0.4b\nheqminer.exe
[2016-11-04 22:44:10] [INFO] [BENCHMARK] Final Speed: 0.000 H/s 
[2016-11-04 22:44:10] [INFO] [BENCHMARK] Benchmark ends
[2016-11-04 22:44:10] [INFO] [FormBenchmark] EndBenchmark() benchmark routine finished
[2016-11-04 22:44:10] [INFO] [CopyBenchmarks] Checking for benchmarks to copy
[2016-11-04 22:44:12] [INFO] [nheqminer-MINER_ID(6)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:12] [INFO] [ExtraLaunchParametersParser] ExtraLaunch params "" for device UUID GPU-8078e1f9-62fe-bee0-ac55-eb3be0e643bb
[2016-11-04 22:44:12] [INFO] [ExtraLaunchParametersParser] Final extra launch params parse ""
[2016-11-04 22:44:13] [INFO] [BENCHMARK] Benchmark starts
[2016-11-04 22:44:13] [INFO] [nheqminer-MINER_ID(6)-DEVICE_IDs(0)] Starting benchmark: -b   -t 0  -cd 0  -cv 1 
[2016-11-04 22:44:13] [INFO] [nheqminer-MINER_ID(6)-DEVICE_IDs(0)] Using miner: bin\nheqminer_v0.4b\nheqminer.exe
[2016-11-04 22:44:15] [INFO] [BENCHMARK] Final Speed: 0.000 H/s 
[2016-11-04 22:44:15] [INFO] [BENCHMARK] Benchmark ends
[2016-11-04 22:44:15] [INFO] [FormBenchmark] EndBenchmark() benchmark routine finished
[2016-11-04 22:44:15] [INFO] [CopyBenchmarks] Checking for benchmarks to copy
[2016-11-04 22:44:17] [INFO] [nheqminer-MINER_ID(7)-DEVICE_IDs(NOT_SET)] NEW MINER CREATED
[2016-11-04 22:44:17] [INFO] [ExtraLaunchParametersParser] ExtraLaunch params "" for device UUID GPU-8078e1f9-62fe-bee0-ac55-eb3be0e643bb
[2016-11-04 22:44:17] [INFO] [ExtraLaunchParametersParser] Final extra launch params parse ""
[2016-11-04 22:44:18] [INFO] [BENCHMARK] Benchmark starts
[2016-11-04 22:44:18] [INFO] [nheqminer-MINER_ID(7)-DEVICE_IDs(0)] Starting benchmark: -b   -t 0  -cd 0  -cv 1 
[2016-11-04 22:44:18] [INFO] [nheqminer-MINER_ID(7)-DEVICE_IDs(0)] Using miner: bin\nheqminer_v0.4b\nheqminer.exe
[2016-11-04 22:44:21] [INFO] [BENCHMARK] Final Speed: 0.000 H/s 
[2016-11-04 22:44:21] [INFO] [BENCHMARK] Benchmark ends
[2016-11-04 22:44:21] [INFO] [FormBenchmark] EndBenchmark() benchmark routine finished
[2016-11-04 22:44:21] [INFO] [CopyBenchmarks] Checking for benchmarks to copy

**Comment by rpcjacobs** [comment]
I **_was_** having the same issue on a 560-Ti. Couldn't get NiceHashMiner to run the equihash benchmark/miner, thus it only used the CPU. Had a look at the logs and tried some manual commands and it seems Cuda 7.5 is is forced for equihash with a hardcoded nheqminer argument. Tried to circumvent that with 'Extra Launch Parameters' but failed to!

I finally got it to work, by building a custom NiceHashMiner which forces 8.0, here's how.

Make the following change https://github.com/nicehash/NiceHashMiner/compare/master...rpcjacobs:patch-1#diff-9b526111082aeccc5e0efafbf908c322
To force the use of Cuda 8.0 instead of 7.5.

Then build it and replace the original .exe with your custom `Release\NiceHashMiner.exe`.
**Be sure to test first if that fixes it**. In my case that wasn't enough, but it might for you.

---

Finally i replaced the following file (which came with the downloads at first startup);
`bin\nheqminer_v0.4b\cuda_tromp.dll` with the one from http://chronx.pl/cuda/cuda_tromp_0.3a.zip supplied by [chronosek](https://github.com/nicehash/nheqminer/issues/82#issuecomment-257429233)

I've uploaded a [drop-in replacement zip](https://12roar.com/shared/crypto/NiceHashMiner_v1.7.3.3_equihash_cuda8.zip), with the above changes for forcing Cuda 8.0 and the new `cuda_tromp.dll`, if you want to test it out and/or don't want to compile it yourself.

Also more info here [nheqminer#82](../../nheqminer/issues/82)
